### ISSUE TITLE
incorrect JavaScript syntax highlighting for quotes in arrays

### ISSUE THREAD

**Issue by cbries** [icsharpcode/AvalonEdit#336]
Commit: up to the latest master variant   (395ef8166870e2c6e1f63a7d97ac22e5e646e790)

Steps to reproduce:
(1) load the AvalonEdit solution
(2) Build & Run the Sample project `ICSharpCode.AvalonEdit.Sample`
(3) switch to "JavaScript" mode
(4) paste the following JavaScript:
```javascript
const texts = [
  `Hello world`,
  `My 'Comma`,
  `My end of Comma'`, 
  `My "Comma`,
  `My incorrect highlighting " Dingens`,
  `Inside "it seems to work" without issue.`
]
```

(5) result
![image](https://user-images.githubusercontent.com/5418062/158761267-cd44c1d8-6e4d-4f47-bc71-38dd294770ba.png)

As you can see the quote highlighting is going over two or more lines.
The JavaScript code is absolutly valid as I see:
![image](https://user-images.githubusercontent.com/5418062/158761403-b29f4a7c-7eb3-4680-b8a4-538e272c6e51.png)

**Comment by siegfriedpammer** [comment]
This is likely because AvalonEdit only has syntax highlighting for variants of JS from the stone age. We will gladly accept a PR that fixes this!

**Comment by cbries** [comment]
Seems to be solved by AvalonEdit v6.3.0.0 and higher:

![image](https://github.com/icsharpcode/AvalonEdit/assets/5418062/6b4995c3-e774-4644-9d14-c94ac042e0c2)
